src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/SqlBulkCopy.cs

```diff
@@ -266,6 +266,31 @@ public SourceColumnMetadata(ValueMethod method, bool isSqlType, bool isDataFeed)
         private DataRowState _rowStateToSkip;
         private IEnumerator _rowEnumerator;
 
+        private int RowNumber
+        {
+            get
+            {
+                int rowNo;
+
+                switch (_rowSourceType)
+                {
+                    case ValueSourceType.RowArray:
+                        rowNo = ((DataTable)_dataTableSource).Rows.IndexOf(_rowEnumerator.Current as DataRow);
+                        break;
+                    case ValueSourceType.DataTable:
+                        rowNo = ((DataTable)_rowSource).Rows.IndexOf(_rowEnumerator.Current as DataRow);
+                        break;
+
+                    case ValueSourceType.DbDataReader:
+                    case ValueSourceType.IDataReader:
+                    case ValueSourceType.Unspecified:
+                    default:
+                        return -1;
+                }
+                return ++rowNo;
```

Comment:
Why do we increment the number?

Reply:
https://github.com/dotnet/SqlClient/pull/437/files/0a2a2194f9ef28e319cc96fb3eab1bdbf9e01dc9#r384217149